Build failure after binutils-dev 2.34

**Describe the bug**

binutils remove those functions, and state that they are not public API, use at your own risk.

https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=950414

```
/home/decster/projects/incubator-doris/be/src/util/bfd_parser.cpp:48:10: error: ‘bfd_get_section_flags’ was not declared in this scope; did you mean ‘bfd_set_section_flags’?
   48 |     if ((bfd_get_section_flags(abfd, sec) & SEC_ALLOC) == 0) {
      |          ^~~~~~~~~~~~~~~~~~~~~
      |          bfd_set_section_flags
/home/decster/projects/incubator-doris/be/src/util/bfd_parser.cpp:51:16: error: ‘bfd_get_section_vma’ was not declared in this scope; did you mean ‘bfd_set_section_vma’?
   51 |     auto vma = bfd_get_section_vma(abfd, sec);
      |                ^~~~~~~~~~~~~~~~~~~
      |                bfd_set_section_vma
/home/decster/projects/incubator-doris/be/src/util/bfd_parser.cpp:55:17: error: ‘bfd_get_section_size’ was not declared in this scope; did you mean ‘bfd_set_section_size’?
   55 |     auto size = bfd_get_section_size(sec);
      |                 ^~~~~~~~~~~~~~~~~~~~
      |                 bfd_set_section_size

```

--- a/be/src/util/bfd_parser.cpp
+++ b/be/src/util/bfd_parser.cpp
@@ -45,14 +45,28 @@ static void find_addr_in_section(bfd* abfd, asection* sec, void* arg) {
     if (ctx->found) {
         return;
     }
+#ifdef bfd_get_section_flags
     if ((bfd_get_section_flags(abfd, sec) & SEC_ALLOC) == 0) {
         return;
     }
+#else
+    if ((bfd_section_flags(sec) & SEC_ALLOC) == 0) {
+        return;
+    }
+#endif
+#ifdef bfd_get_section_vma
     auto vma = bfd_get_section_vma(abfd, sec);
+#else
+    auto vma = bfd_section_vma(sec);
+#endif
     if (ctx->pc < vma) {
         return;
     }
+#ifdef bfd_get_section_size
     auto size = bfd_get_section_size(sec);
+#else
+    auto size = bfd_section_size(sec);
+#endif
     if (ctx->pc >= vma + size) {
         return;
     }

